solr/prometheus-exporter/src/test/org/apache/solr/prometheus/scraper/SolrStandaloneScraperBasicAuthTest.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.prometheus.scraper;
+
+import io.prometheus.client.Collector;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import org.apache.lucene.tests.util.LuceneTestCase;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.client.solrj.impl.Http2SolrClient;
+import org.apache.solr.common.util.ExecutorUtil;
+import org.apache.solr.common.util.IOUtils;
+import org.apache.solr.common.util.SolrNamedThreadFactory;
+import org.apache.solr.prometheus.PrometheusExporterTestBase;
+import org.apache.solr.prometheus.exporter.MetricsConfiguration;
+import org.apache.solr.prometheus.exporter.PrometheusExporterSettings;
+import org.apache.solr.prometheus.exporter.SolrClientFactory;
+import org.apache.solr.prometheus.exporter.SolrScrapeConfiguration;
+import org.apache.solr.prometheus.utils.Helpers;
+import org.apache.solr.util.SolrJettyTestRule;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class SolrStandaloneScraperBasicAuthTest extends SolrTestCaseJ4 {
+
+  @ClassRule public static final SolrJettyTestRule solrRule = new SolrJettyTestRule();
+
+  private static Http2SolrClient solrClient;
+  private static MetricsConfiguration configuration;
+  private static SolrStandaloneScraper solrScraper;
+  private static ExecutorService executor;
+
+  private static String user = "solr";
+  private static String pass = "SolrRocks";
+
+  @BeforeClass
+  public static void setupSolrHome() throws Exception {
+    Path solrHome = LuceneTestCase.createTempDir();
+    Files.copy(
+        SolrTestCaseJ4.TEST_PATH().resolve("security.json"), solrHome.resolve("security.json"));
+    solrRule.startSolr(solrHome);
+
+    Path configSet = LuceneTestCase.createTempDir();
+    SolrStandaloneScraperTest.createConf(configSet);
+    solrRule
+        .newCollection()
+        .withConfigSet(configSet.toString())
+        .withBasicAuthCredentials(user, pass)
+        .create();
+
+    configuration =
+        Helpers.loadConfiguration("conf/prometheus-solr-exporter-scraper-test-config.xml");
+
+    PrometheusExporterSettings settings = PrometheusExporterSettings.builder().build();
+    SolrScrapeConfiguration scrapeConfiguration =
+        SolrScrapeConfiguration.standalone(solrRule.getBaseUrl())
+            .withBasicAuthCredentials(user, pass);
+    solrClient =
+        new SolrClientFactory(settings, scrapeConfiguration)
+            .createStandaloneSolrClient(solrRule.getBaseUrl());
+    executor =
+        ExecutorUtil.newMDCAwareFixedThreadPool(
+            25, new SolrNamedThreadFactory("solr-cloud-scraper-tests"));
+    solrScraper = new SolrStandaloneScraper(solrClient, executor, "test");
+
+    Helpers.indexAllDocs(solrClient);
```

Comment:
don't love this class name?  Helpers feels very generic...   Also, we don't have the equivaletn of indexAllDocs somewhere already?

Reply:
Actually, the class name I didn't love was `Helpers`...  ;-)   I feel like that method `indexAllDocs` must exist already!

Reply:
ah right. this was already there. I picked up the SolrStandaloneScraperTest test (cleaned it up a bit) and added basic auth to it. `indexAllDocs` will actually index a local copy of the solr examples files. I'm pretty sure this was not strictly needed for this, but yeah I didn't add any of it.